src/permissions/Caveat.ts

```diff
@@ -136,6 +120,31 @@ export type CaveatSpecificationConstraint = {
   validator?: CaveatValidator<any>;
 };
 
+export type RestrictedMethodCaveatSpecificationConstraint =
+  CaveatSpecificationBase & {
+    /**
+     * The decorator function used to apply the caveat to restricted method
+     * requests.
+     */
+    decorator: CaveatDecorator<any>;
+  };
+
+export type EndowmentCaveatSpecificationConstraint = CaveatSpecificationBase;
+
+/**
+ * The constraint for caveat specification objects. Every {@link Caveat}
+ * supported by a {@link PermissionController} must have an associated
+ * specification, which is the source of truth for all caveat-related types.
+ * In addition, a caveat specification may include a decorator function used
+ * to apply the caveat's attenuation to a restricted method. It may also include a validator
+ * function specified by the consumer.
```

Comment:
```suggestion
 * to apply the caveat's attenuation to a restricted method. It may also include
 * a validator function specified by the consumer.
```